lib/r10k/action/deploy/environment.rb

```diff
@@ -215,7 +215,7 @@ def write_environment_info!(environment, started_at, success)
           # make this file write as atomic as possible in pure ruby
           final   = "#{environment.path}/.r10k-deploy.json"
           staging = "#{environment.path}/.r10k-deploy.json~"
-          File.open(staging, 'w') do |f|
+          File.open(staging, 'wb') do |f|
```

Comment:
why is it necessary to write this file as binary?

Reply:
It may not be necessary; I was planning on confirming if this was required before creating any PR back to source project. We'd had issues with writing files on Windows in other projects when write mode wb isn't use. 
See: https://stackoverflow.com/questions/3682359/what-are-the-ruby-file-open-modes-and-options

Binary mode also supresses CRLF line endings on Windows. When ruby was writing files out to disk on Windows CRLF was being added to line ending, causing those scripts to fail when copied to Nix systems by bolt.